src/librustc_resolve/lib.rs

```diff
@@ -3352,39 +3350,24 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         NoSuggestion
     }
 
-    fn find_best_match_for_name(&mut self, name: &str) -> SuggestionType {
-        let mut maybes: Vec<token::InternedString> = Vec::new();
-        let mut values: Vec<usize> = Vec::new();
-
+    fn find_best_match(&mut self, name: &str) -> SuggestionType {
         if let Some(macro_name) = self.session.available_macros
-                                 .borrow().iter().find(|n| n.as_str() == name) {
+                                  .borrow().iter().find(|n| n.as_str() == name) {
             return SuggestionType::Macro(format!("{}!", macro_name));
         }
 
+        let mut names: Vec<Name> = Vec::new();
```

Comment:
We don't need a vector or any pushes here. This can be constructed via a map iterator (`self.value_ribs.iter().rev().map(|(&k, _)| k)`) and passed directly to `find_best_match_for_name()`